server/src/test/java/org/apache/druid/client/BrokerInternalQueryConfigTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.druid.segment.TestHelper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BrokerInternalQueryConfigTest
+{
+  private static final ObjectMapper MAPPER = TestHelper.makeJsonMapper();
+
+  @Test
+  public void testSerde() throws Exception
```

Comment:
Can you add a test for an improperly configured map.

If a user configures the context incorrectly, for example `{` Is the expected behavior that the broker will fail to start?

Reply:
Expected behavior is good question. I guess I'd personally be okay with either. It may be less confusing to an operator if the service just didn't start. Otherwise we could have some explicit WARN logging and startup with a null context. Which do you think would be best?

Reply:
I think making the service fail to start would be preferred.

In this case a warn log would likely be ignored since you wouldn't really have a need to check the logs until someone notices something strange with the priority of the generated metadata queries - which I think would be very hard to debug.

Reply:
I agree that failure to startup is a fine choice for this. I added a test for the malformed config as well as an additional test for default behavior using a Guice injector.